crates/bevy_audio/src/audio_output.rs

```diff
@@ -74,8 +89,101 @@ where
     let world = world.cell();
     let audio_output = world.get_non_send::<AudioOutput<Source>>().unwrap();
     let mut audio = world.get_resource_mut::<Audio<Source>>().unwrap();
+    let mut sinks = world.get_resource_mut::<Assets<AudioSink>>().unwrap();
 
     if let Some(audio_sources) = world.get_resource::<Assets<Source>>() {
-        audio_output.try_play_queued(&*audio_sources, &mut *audio);
+        audio_output.try_play_queued(&*audio_sources, &mut *audio, &mut *sinks);
     };
 }
+
+/// Asset controlling the playback of a sound
+///
+/// ```
+/// # use bevy_ecs::system::{Local, Res};
+/// # use bevy_asset::{Assets, Handle};
+/// # use bevy_audio::AudioSink;
+/// // Execution of this system should be controlled by a state or input,
+/// // otherwise it would just toggle between play and pause every frame.
+/// fn pause(
+///     audio_sinks: Res<Assets<AudioSink>>,
+///     music_controller: Local<Handle<AudioSink>>,
+/// ) {
+///     if let Some(sink) = audio_sinks.get(&*music_controller) {
+///         if sink.is_paused() {
+///             sink.play()
+///         } else {
+///             sink.pause()
+///         }
+///     }
+/// }
+/// ```
```

Comment:
I don't think that it's needed: the only thing new in this system is the kind of asset, but systems with those parameters are already tested elsewhere. and if `AudioSink` is not a valid asset, it would still fail to compile